cumulusci/cli/cci.py

```diff
@@ -1425,83 +1425,235 @@ def task_info(runtime, task_name):
+            # Values come in as None when they are specified via a cumulusci.yml file
+            for name, value in kwargs.items():
+                if value and name in task_options.keys():
+
+                    # Add options if they aren't present on this task
+                    # they are defined on the parent class
+                    if "options" not in task_config.config:
+                        task_config.config["options"] = {}
+                    task_config.config["options"][name] = value
+
+            try:
+                task = task_class(
+                    task_config.project_config, task_config, org_config=org_config
                 )
 
-            # Override the option in the task config
-            task_config.config["options"][name] = value
+                if kwargs.pop("debug_before", None):
+                    self._import_pdb_and_set_trace()
 
-    # Create and run the task
-    try:
-        task = task_class(
-            task_config.project_config, task_config, org_config=org_config
-        )
+                task()
 
-        if debug_before:
-            import pdb
+                if kwargs.pop("debug_after", None):
+                    self._import_pdb_and_set_trace()
 
-            pdb.set_trace()
+            finally:
+                RUNTIME.alert(f"Task complete: {task_name}")
 
-        task()
+        return click.Command(task_name, params=params, callback=run_task)
 
-        if debug_after:
-            import pdb
+    def resolve_command(self, ctx, args):
+        """
+        CumulusCI overrides this method from the MultiCommand
+        class to allow us access to the raw command line
+        args being passed in. This allows us to convert
+        the old option syntax to the new option syntax.
+        """
+        args = self._convert_old_option_syntax(args)
+        return click.MultiCommand.resolve_command(self, ctx, args)
 
-            pdb.set_trace()
-    finally:
-        runtime.alert(f"Task complete: {task_name}")
+    def _convert_old_option_syntax(self, args):
+        """
+        Given a list of args, convert any options specified
+        with the old format `-o name value` to the new format `--name value`.
+
+        Args:
+            param1: The list of arguments to convert
+
+        Returns:
+            The list of arguments
+
+        Raises:
+            CumulusCIUsageError: if there is an error in parsing the old option syntax,
+            or the option doesn't exist for the given task.
+        """
+        duplicate = self._has_duplicate_options(args)
+        if duplicate:
+            raise CumulusCIUsageError(
+                f"Found duplicate option `{duplicate}` in given command:\n cci task run {' '.join(args)}"
+            )
+
+        task_name = args[0]
+        while "-o" in args:
+            idx = args.index("-o")
+            opt_name = args[idx + 1]
+            opt_value = args[idx + 2]
```

Comment:
I tried running the util_sleep task with just `-o seconds` (i.e. forgot the value), and got an IndexError here.

Reply:
I notice you have code for looping through the args and identifying the options both here and in _parse_option_names. Why not have one method which does the parsing and yields name/value pairs, and another which iterates through that, identifies duplicates (by adding the names to a set and checking for existence in that set), and adds the modified args to a new list.